tooling/nargo/src/ops/foreign_calls.rs

```diff
@@ -120,9 +120,11 @@ impl DefaultForeignCallExecutor {
         decode_string_value(&fields)
     }
 
-    fn execute_println(foreign_call_inputs: &[ForeignCallParam]) -> Result<(), ForeignCallError> {
+    fn execute_print(foreign_call_inputs: &[ForeignCallParam]) -> Result<(), ForeignCallError> {
         let display_values: PrintableValueDisplay = foreign_call_inputs.try_into()?;
-        println!("{display_values}");
+        let skip_newline = foreign_call_inputs[1].unwrap_value().is_zero();
+        print!("{display_values}{}", if skip_newline { "" } else { "\n" });
```

Comment:
Using `foreign_call_inputs[1]` here gives the incorrect result if a struct is printed. E.g. `print(Foo { x: 1, y: 1 });` will always print with a newline because `y = 1` despite `print` being used.
```suggestion
        let last_input = foreign_call_inputs.last().expect("print expected at least 1 argument");
        let skip_newline = last_input.unwrap_value().is_zero();
        print!("{display_values}{}", if skip_newline { "" } else { "\n" });
```